core/src/avm1/globals/point.rs

```diff
@@ -0,0 +1,376 @@
+        .get("y", avm, context)?
+        .resolve(avm, context)?
+        .as_number(avm, context)?;
+    let other = value_to_point(
+        args.get(0).unwrap_or(&Value::Undefined).to_owned(),
+        avm,
+        context,
+    )?;
+    let object = point_to_object((this_x - other.0, this_y - other.1), avm, context)?;
+    Ok(object.into())
+}
+
+fn distance<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    if args.len() < 2 {
+        return Ok(Value::Undefined.into());
+    }
+
+    let a = value_to_point(
+        args.get(0).unwrap_or(&Value::Undefined).to_owned(),
+        avm,
+        context,
+    )?;
+    let b = value_to_point(
+        args.get(1).unwrap_or(&Value::Undefined).to_owned(),
+        avm,
+        context,
+    )?;
+    let xd = a.0 - b.0;
+    let yd = a.1 - b.1;
+    let distance = (xd * xd + yd * yd).sqrt();
+    Ok(distance.into())
+}
+
+fn polar<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let length = args
+        .get(0)
+        .unwrap_or(&Value::Undefined)
+        .as_number(avm, context)?;
+    let angle = args
+        .get(1)
+        .unwrap_or(&Value::Undefined)
+        .as_number(avm, context)?;
+    let point = point_to_object((length * angle.cos(), length * angle.sin()), avm, context)?;
+    Ok(point.into())
+}
+
+fn interpolate<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    if args.len() < 3 {
+        return Ok(Value::Undefined.into());
+    }
+
+    let a = value_to_point(args.get(0).unwrap().to_owned(), avm, context)?;
+    let b = value_to_point(args.get(1).unwrap().to_owned(), avm, context)?;
+    let f = args.get(2).unwrap().as_number(avm, context)?;
+    let result = (b.0 - (b.0 - a.0) * f, b.1 - (b.1 - a.1) * f);
+    Ok(point_to_object(result, avm, context)?.into())
+}
+
+fn to_string<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let x = this
+        .get("x", avm, context)?
+        .resolve(avm, context)?
+        .coerce_to_string(avm, context)?;
+    let y = this
+        .get("y", avm, context)?
+        .resolve(avm, context)?
+        .coerce_to_string(avm, context)?;
+
+    Ok(format!("(x={}, y={})", x, y).into())
+}
+
+fn length<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    let point = value_to_point(this.into(), avm, context)?;
+    let length = (point.0 * point.0 + point.1 * point.1).sqrt();
+    Ok(length.into())
+}
+
+fn normalize<'gc>(
```

Comment:
Some ones that are currently incorrect:

`(NaN, 0).normalize()`
`(undefined, 0).normalize(1)`
`(10, null).normalize(0)`